include/caffe/net.hpp

```diff
@@ -89,7 +89,7 @@ class Net {
    * @brief For an already initialized net, implicitly copies (i.e., using no
    *        additional memory) the pre-trained layers from another Net.
    */
-  void ShareTrainedLayersWith(Net* other);
+  void ShareTrainedLayersWith(Net* other) const;
```

Comment:
Oops, I was thinking this went the other way, thanks for pointing that out. I guess there is some subtle reason why the compiler does not generate any warnings in this case. I'll push a corrected version shortly...